libp2p/switch.nim

```diff
@@ -244,7 +244,9 @@ proc upgradeIncoming(s: Switch, incomingConn: Connection) {.async, gcsafe.} = #
       await ms.handle(cconn)
     except CatchableError as exc:
       debug "Exception in secure handler during incoming upgrade", msg = exc.msg, conn
-      if not cconn.upgraded.finished:
+      if  not isNil(cconn) and
```

Comment:
Ah, yeah - I guess, if the peer/connection is being cleared up, this can become `Nil` at any time. Not fun, but a connection is a shared resource, so this is probably unavoidable. 

Maybe, we should hide the upgraded future behind a property and do the checks there instead of the ugly `and`? Don't have to be this PR, can be cleaned up later.

Reply:
also, might be a good idea to cherry pick into its own PR.

Reply:
yeah I asked if this should be cherry picked but not much interest, I'm going to merge this today anyway I think!

Reply:
still not sure about the nil to be fair, just at this point better be safe than sorry :D ?